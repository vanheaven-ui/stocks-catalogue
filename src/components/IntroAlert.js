import { useState } from 'react';
import { Alert, Button } from 'react-bootstrap';

const IntroAlert = () => {
  const [show, setShow] = useState(true);
  return (
    <>
      { show && (
      <Alert variant="primary" onClose={() => setShow(false)} dismissible>
        <Alert.Heading id="heading">
          <span role="img" aria-labelledby="heading">ðŸ‘‹</span>
          Hey!, Welcome to stocks catalogue
        </Alert.Heading>
        <hr />
        <p style={{ fontSize: 16 }}>
          This is a Single Page Application(SPA) that consumes the
          {' '}
          <a
            href="https://financialmodelingprep.com/developer/docs/"
            target="_blank"
            rel="noreferrer"
            style={{ textDecoration: 'underline', fontWeight: 700 }}
          >
            financialmodelingprep API
            {' '}
          </a>
          to get real time stock trading details and news so that they are easily accessible.
          <br />
          <strong>Remember to use the filter to narrow down your browsing list.</strong>
          <br />
          Enjoy and make the best decision in this volatile business!
        </p>
      </Alert>
      )}
      { !show && (
      <Button variant="light" onClick={() => setShow(true)} id="greeting">
        <span role="img" aria-labelledby="greeting">ðŸ‘‹</span>
        Greeting!
      </Button>
      )}
    </>
  );
};

export default IntroAlert;
