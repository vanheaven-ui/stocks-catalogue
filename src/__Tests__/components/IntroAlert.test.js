import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {
  cleanup, fireEvent, render,
} from '@testing-library/react';
import IntroAlert from '../../components/IntroAlert';
import rootNode from '../helpers';
import '@testing-library/jest-dom/extend-expect';

describe('IntroALert component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    ReactDOM.render(<IntroAlert />, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const tree = renderer.create(<IntroAlert />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct element', () => {
    const { getByRole } = render(<IntroAlert />);
    expect(getByRole('button')).toBeInTheDocument();
  });

  test('button click closes the alert', () => {
    const { getByRole, getByText } = render(<IntroAlert />);
    fireEvent.click(getByRole('button'));
    expect(getByText('Greeting!')).toBeInTheDocument();
  });
});
