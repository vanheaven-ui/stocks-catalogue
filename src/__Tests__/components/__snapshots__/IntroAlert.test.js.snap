// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroALert component doesnot change unexpectedly 1`] = `
<div
  className="fade alert alert-primary alert-dismissible show"
  role="alert"
>
  <button
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
    <span
      className="sr-only"
    >
      Close alert
    </span>
  </button>
  <div
    className="alert-heading h4"
    id="heading"
  >
    <span
      aria-labelledby="heading"
      role="img"
    >
      ðŸ‘‹
    </span>
    Hey!, Welcome to stocks catalogue
  </div>
  <hr />
  <p
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    This is a Single Page Application(SPA) that consumes the
     
    <a
      href="https://financialmodelingprep.com/developer/docs/"
      rel="noreferrer"
      style={
        Object {
          "fontWeight": 700,
          "textDecoration": "underline",
        }
      }
      target="_blank"
    >
      financialmodelingprep API
       
    </a>
    to get real time stock trading details and news so that they are easily accessible.
    <br />
    <strong>
      Remember to use the filter to narrow down your browsing list.
    </strong>
    <br />
    Enjoy and make the best decision in this volatile business!
  </p>
</div>
`;
